src/sidecar/connections/watcherUtils.test.ts

```diff
@@ -120,6 +125,21 @@ describe("connectionEventHandler", () => {
         }),
         true,
       );
+
+      // directConnectionCreatedStub should be called only for CREATED events
+      if (action === ConnectionEventAction.CREATED) {
+        assert.strictEqual(
+          directConnectionCreatedStub.calledOnce,
+          true,
+          "directConnectionCreatedStub called",
+        );
+      } else {
+        assert.strictEqual(
+          directConnectionCreatedStub.notCalled,
+          true,
+          "directConnectionCreatedStub not called",
+        );
```

Comment:
Same here, minus the messages since the sinon assertions API makes it really nice when the assertions fail
```suggestion
        sinon.assert.calledOnce(directConnectionCreatedStub.calledOnce);
      } else {
        sinon.assert.notCalled(directConnectionCreatedStub);
```